benchmark/fs/readfile-partitioned.js

```diff
@@ -0,0 +1,84 @@
+// Call fs.readFile over and over again really fast.
+// Then see how many times it got called.
+// Also report how many times we were able to run a small concurrent
+// operation on the threadpool.
+'use strict';
+
+const path = require('path');
+const common = require('../common.js');
+const filename = path.resolve(__dirname,
+                              `.removeme-benchmark-garbage-${process.pid}`);
+const fs = require('fs');
+const zlib = require('zlib');
+const assert = require('assert');
+
+const bench = common.createBenchmark(main, {
+  dur: [5],
+  len: [1024, 16 * 1024 * 1024],
+  concurrent: [1, 10]
+});
+
+function main(conf) {
+  const len = +conf.len;
+  try { fs.unlinkSync(filename); } catch (e) {}
+  var data = Buffer.alloc(len, 'x');
+  fs.writeFileSync(filename, data);
+  data = null;
+
+  var zipData = Buffer.alloc(1024, 'a');
+
+  var reads = 0;
+  var zips = 0;
+  var benchEnded = false;
+  bench.start();
+  setTimeout(function() {
+    const totalOps = reads + zips;
+    console.log('bench ended: ' +
+      `reads ${reads}, zips ${zips}, total ops ${totalOps}`);
```

Comment:
Can you remove this?

Reply:
This log emits the relative throughput of the large `fs.readFile` and the compression operation. I initially tried two `bench.end` calls but @mscdex [indicated](https://github.com/nodejs/node/pull/17054#discussion_r151430475) that this was improper. I don't think calling a single `bench.end` with the combined `totalOps` fairly describes the situation because there are far more zips than reads.

Got an alternative?

Reply:
This is not compliant with how we do the statistical comparison. Maybe @AndreasMadsen has a better idea.

Reply:
I'm happy to change to whatever is appropriate. Let me know.

Reply:
I'm not sure I understand what is going on (I haven't walked through the entire PR). – That being said, it appears that `benchmark/fs/readfile.js` is already measuring something equivalent to `reads`. What this measure is the number of small concurrent operation (`zips`). I think we can just report this number, right?

Reply:
Yes, we could report only zips, and this would be better than nothing.

Two possible downsides for your consideration:
1. We lose the information about the effect of the zips on the partitioned readFile tasks. We don't get this information from `readfile.js` because there are no concurrent small tasks there.

2. I'm a bit hesitant to rely on separate benchmarks, `readfile.js` for `fs.readFile` performance and `readfile-partitioned.js` for "concurrent small job" performance. Then the `bench` details have to be kept in sync across the two benchmarks.

> This is not compliant with how we do the statistical comparison.

I don't understand why the current code is undesirable. Are there guidelines somewhere about this?

Reply:
> I don't understand why the current code is undesirable. Are there guidelines somewhere about this?

@davisjam Without having looked at everything you have done, your approach to quantitatively determining if there is a difference between the current implementation and this implementation does not use statistic. It simply eyeballs some numbers.

We have a statistical tool for comparing benchmark results across versions. It is documented here: https://github.com/nodejs/node/blob/master/doc/guides/writing-and-running-benchmarks.md#comparing-nodejs-versions

This eliminates the human tendency to see patterns where there are none and to see randomness where there are patterns.

This tool only allows for comparing one value per benchmark. Yes, this is a limitation of the tool but it also makes the benchmarks very easy to understand for the user. Likewise, all our benchmarks are made such that higher numbers are better numbers, this is also to make them easier to understand.

We have also documented how to write the benchmarks in:
https://github.com/nodejs/node/blob/master/doc/guides/writing-and-running-benchmarks.md#basics-of-a-benchmark

Reply:
>  It simply eyeballs some numbers.

I appreciate your desire for rigor. I personally have no problem eyeballing numbers that are [two orders of magnitude](https://github.com/nodejs/node/pull/17054#issuecomment-345133363) apart ;-).

> This tool only allows for comparing one value per benchmark.

Right. So I can simply report the `zip` throughput, and in conjunction with the `readfile.js` this will tell us the throughput improvement for small jobs and the throughput degradation for large jobs.

@mcollina @AndreasMadsen If you are OK with the possible issues with this approach that I discussed above then I will change the benchmark to only report the zip throughput.

Reply:
Addressed in 60bdd89.